opslevel/resource_opslevel_trigger_definition.go

```diff
@@ -80,16 +80,35 @@ func resourceTriggerDefinition() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validation.StringInSlice(opslevel.AllCustomActionsEntityTypeEnum, false),
 			},
+			"extended_team_access": {
+				Type:        schema.TypeList,
+				Description: "The set of additional teams who can invoke this Trigger Definition.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
 		},
 	}
 }
 
 func resourceTriggerDefinitionCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	var extended_teams *[]opslevel.IdentifierInput
+
+	extendedTeamAccessArray := getStringArray(d, "extended_team_access")
+	if len(extendedTeamAccessArray) > 0 {
+		my_teams := []opslevel.IdentifierInput{}
+		for _, team := range extendedTeamAccessArray {
+			my_teams = append(my_teams, *opslevel.NewIdentifier(team))
+		}
+		extended_teams = &my_teams
+	}
+
 	input := opslevel.CustomActionsTriggerDefinitionCreateInput{
-		Name:          d.Get("name").(string),
-		Owner:         *opslevel.NewID(d.Get("owner").(string)),
-		Action:        *opslevel.NewID(d.Get("action").(string)),
-		AccessControl: opslevel.CustomActionsTriggerDefinitionAccessControlEnum(d.Get("access_control").(string)),
+		Name:               d.Get("name").(string),
+		Owner:              *opslevel.NewID(d.Get("owner").(string)),
+		Action:             *opslevel.NewID(d.Get("action").(string)),
+		AccessControl:      opslevel.CustomActionsTriggerDefinitionAccessControlEnum(d.Get("access_control").(string)),
+		ExtendedTeamAccess: extended_teams,
 	}
```

Comment:
```suggestion
	}
	if len(extended_teams) > 0 {
	    input.ExtendedTeamAccess = &extended_teams
	}
```

I'm not married to it but it moves the logic for the respective parts closer to where they are used.

Reply:
This is _way_ better! I started with the suggestion above and was just missing this if statement 🙌